lua/luasnip/util/environ.lua

```diff
@@ -125,6 +125,19 @@ local fake_var = {
 
     __tostring = function(obj)
         return obj[1]
+    end,
+    __index = function(obj, key)
+        local wrapped_str = obj[1]
+        local val = wrapped_str[key]
+        if type(val) == "function" then
+            return function (x, ...)
+                if x == obj then
+                    x = wrapped_str
+                end
+                return string[key](x, ...)
```

Comment:
Indexing val instead of immediately `string` is a nice idea :+1:
I think `val` instead of `string[key]` would be slightly more coherent, wdyt?

Reply:
Also +1 for the `x==obj`-check, that might prevent some (okay, unlikely, but better to be safe here) errors.